homeassistant/components/light/deconz.py

```diff
@@ -0,0 +1,181 @@
+"""
+Support for deCONZ light support.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/light/deconz/
+"""
+
+import asyncio
+import logging
+
+from homeassistant.components.deconz import (
+    ATTR_FW_VERSION, ATTR_MANUFACTURER, ATTR_MODEL_ID, ATTR_UNIQUE_ID,
+    DECONZ_DATA, DOMAIN)
+from homeassistant.components.light import (
+    Light, ATTR_BRIGHTNESS, ATTR_FLASH, ATTR_COLOR_TEMP, ATTR_EFFECT,
+    ATTR_RGB_COLOR, ATTR_TRANSITION, EFFECT_COLORLOOP, FLASH_LONG, FLASH_SHORT,
+    SUPPORT_BRIGHTNESS, SUPPORT_COLOR_TEMP, SUPPORT_EFFECT, SUPPORT_FLASH,
+    SUPPORT_RGB_COLOR, SUPPORT_TRANSITION, SUPPORT_XY_COLOR)
+from homeassistant.core import callback
+from homeassistant.util.color import color_RGB_to_xy
+
+
+DEPENDENCIES = [DOMAIN]
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_LIGHT_GROUP = 'LightGroup'
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup light platform for Deconz."""
+    lights = hass.data[DECONZ_DATA].lights
+    groups = hass.data[DECONZ_DATA].groups
+
+    for _, light in lights.items():
+        async_add_devices([DeconzLight(light)], True)
+
+    for _, group in groups.items():
+        if group.lights:  # Don't create entity for group not containing light
+            async_add_devices([DeconzLight(group)], True)
```

Comment:
Collect all entities in a list and call `async_add_devices` once.

Reply:
Done